app/permissions/PermissionsActionCheck.scala

```diff
@@ -27,6 +27,56 @@ trait PermissionActionFilter extends ActionFilter[UserRequest] with Logging {
     }
   }
 }
+trait ModifyCollectionsPermissionsCheck extends Logging { self: BaseFaciaController =>
+
+  def acl: Acl
+  def configAgent: ConfigAgent
+
+  private def testAccess(email: String, priorities: Set[PermissionsPriority], isLaunch: Boolean) = {
+
+    if (priorities.isEmpty) AccessGranted
+    else if (isLaunch)
+      acl.testUserGroupsAndCollections(
+        Permissions.LaunchEditorialFronts, Permissions.LaunchCommercialFronts,
+        Permissions.FrontsAccess, "facia-tool-allow-launch-editorial-fronts-for-all")(email, priorities)
+    else
+      acl.testUserGroupsAndCollections(
+        Permissions.EditEditorialFronts, Permissions.LaunchCommercialFronts,
+        Permissions.FrontsAccess, "facia-tool-allow-edit-editorial-fronts-for-all")(email, priorities)
+  }
+
+  def withModifyGroupPermissionForCollections[A](priorities: Set[PermissionsPriority], secondaryPriorities: Set[PermissionsPriority],
+                                                 isLaunch: Boolean = false)(block: => Future[Result])
+                                                (implicit request: UserRequest[A],
+                                                 executionContext: ExecutionContext): Future[Result] = {
+
+    (testAccess(request.user.email, priorities, isLaunch), testAccess(request.user.email, secondaryPriorities, isLaunch)) match {
+      case (AccessGranted, AccessGranted) => block
+      case _ => {
+        logger.warn(s"User with e-mail ${request.user.email} not authorized to modify collection with priorities $priorities")
```

Comment:
Can we also include the permissions that the user does have? That has helped me debug issues before